lib/ex_ice/ice_agent/impl.ex

```diff
@@ -235,17 +235,39 @@ defmodule ExICE.ICEAgent.Impl do
   def add_remote_candidate(ice_agent, remote_cand) do
     Logger.debug("New remote candidate: #{inspect(remote_cand)}")
 
-    case Candidate.unmarshal(remote_cand) do
-      {:ok, remote_cand} ->
-        ice_agent = do_add_remote_candidate(ice_agent, remote_cand)
-        Logger.debug("Successfully added remote candidate.")
+    resolve_address = fn
+      remote_cand when is_binary(remote_cand.address) ->
```

Comment:
I'm wondering why this isn't just a private function?

Reply:
I put it here as it is very specific to the place where it is used